roles/pool/config-examples/pool-config-local-tp-example.toml

```diff
@@ -24,5 +24,8 @@ pool_signature = "Stratum v2 SRI Pool"
 # Template Provider config
 # Local TP (this is pointing to localhost so you must run a TP locally for this configuration to work)
 tp_address = "127.0.0.1:8442"
+# You'll need to get tp_authority_public_key from the logs of your TP, for example:
```

Comment:
should it be like this here?
`tp_authority_public_key = ""`
Also, the following comment "You'll need to get tp_authority_public_key from the logs of your TP" is against what I understood to have agreed: if you run a local TP yuu clearly trust it and there is no need of certificate authority (and no need to retrive that from TP logs)

Reply:
In this example the TP is local, but it doesn't have to be. Maybe change "You'll need to" to "You may" and mention that it's not necessary for a TP on localhost.

Reply:
made some changes in https://github.com/stratum-mining/stratum/pull/752/commits/1a48fa563fe70bea4c9f3dc34b7e7ecf1d9254c3

lmk what you guys think

Reply:
Not sure if it make sense as a comment. You know the auth pub key if you see it in the log and TP is local it means that you set the TP with the auth pub key so you have to know it. I think that this comment is can confuse the reader.

Reply:
>  it means that you set the TP with the auth pub key so you have to know it.

The TP generates its authority key automatically, so it's not true that the user set it.

Reply:
yep but that is not how it should work? we decided to have a way to communicate the key to the process via config or whatever,
Having  the TP generating it do not make any sense to me

Reply:
But then the user needs a separate tool to generate an authority key. It seems easier to generate one automatically.

I think only a mining pool needs a way to use the same authority key for both the template provider and pool. My plan is to support that use case by adding a `-sv2cert` flag where the user can pass in the certificate. The TP won't have to touch the authority private key. For that use case we also need a way to sign the certificate. This is why the TP prints its static key to the log.

Reply:
The authority key is used by some role in order to be trusted. A self generated certified authority key it like "trust me because I am certified by myself", which is a bit pointless.  Perhaps there are other use cases that I am not aware of.

Reply:
> A self generated certified authority key it like "trust me because I am certified by myself"

It's for the use case of someone else wanting to connect to my template provider.

Reply:
If is the application itself that generates the authority key, then you have no way to find out that you are communicating with the correct endpoint. If the TP is third party, then you must ask the institution that hosts it the authority key. If it is not, than you run it yourself and you don't need any certificate.

I like the solution `-sv2cert`, with the authority key as input. If I understand it correctly, you must then communicate this key privately to every client that wants to connect to your TP, which is more or less what I wrote above.

Reply:
In the above scenario if Alice's bitcoind go down she must resend the key to Bob. I really can not imagine a scenario where one want to provide a TP service but is ok in having randomly generated auth key that change everytime that bitcoind is respwaned.

Reply:
> bitcoind go down she must resend the key to Bob.

Bitcoin Core stores the key. Only Alice deletes the data directory, and not have a backup, would she have to generate a new key.

Reply:
ok this make sense to me

Reply:
Yes but there is no need to retrieve is from the log every time. It is better to let decide the user to decide whenever to use a new authority key. I would say also that could be a little annoying to do it manually. I like the `-sv2cert` solution btw!

Reply:
Some messages didn't appear to me at the moment of writing my last message. The idea of generating the key and storing it seems good to me and I ack it. In order to reuse an older key (from a previous backup) we should still be able to pass it as an argument to the `bitcoind`, like the `-sv2cert` idea.